toolchain/base/BUILD

```diff
@@ -22,3 +22,19 @@ cc_library(
         "@llvm-project//llvm:Support",
     ],
 )
+
+cc_library(
+    name = "struct_reflection",
+    hdrs = ["struct_reflection.h"],
+)
+
+cc_test(
+    name = "struct_reflection_test",
+    size = "small",
+    srcs = ["struct_reflection_test.cpp"],
+    deps = [
+        ":struct_reflection",
+        "//testing/base:gtest_main",
+        "@com_google_googletest//:gtest",
+    ],
```

Comment:
Nothing really toolchain-specific here, how about putting this in //common?

(happy for that to be a separate PR or something if it makes merges or other stuff hard)

Reply:
Done.